typescript/src/server/rxjs-server.ts

```diff
@@ -1,81 +1,63 @@
 import WebSocket from "ws";
 import Socket from "./rxjs-socket";
-import { BehaviorSubject, filter, map, Observable, Observer, scan, Subject } from "rxjs";
+import { Observable } from "rxjs";
+import { once } from "events";
 
 export interface Server {
-    listening: BehaviorSubject<boolean>;
-    close(): void;
-    on(): Observable<[Socket, Socket]>;
+  listening: Promise<any>;
+  close(): void;
+  on(): Observable<[Socket, Socket]>;
 }
 
-export default class ServerImpl implements Server {
-    private subject: Subject<[Socket, Socket]>;
```

Comment:
The `Subject` here was completely removed in favor of a function that just returns an Observable. GIven than the `server` instance is external to the observable, it's multicast by default... but that turned out not to matter, as only one instance was ever spun up.

`Subjects` are notably slower and heavier than observables, they have internal state, etc. They should be avoided as a means of just creating an observable (which is what was being done here). Observables, on the other hand, are [just fancy functions with callbacks](https://benlesh.com/posts/learning-observable-by-building-observable/#observables-are-just-functions)